README.md

```diff
@@ -1 +1,106 @@
+```java
+client.<endpoint>.<operation>(parameters);
+```
+On the client object, you chain the endpoint name (for example, collections), and then chain the operation name 
+(createCollection). 
+
+### What is being returned
+All methods return a `Response` object, which holds the HTTP status code and the body of the response (as JSON). Can be 
+retrieved by using `.getHttpStatusCode()` and `.getBody()`, respectively. You can parse the JSON response body using your 
+preferred JSON parsing library or method.
+
+### Optional Fields
+If the operation request contains one optional field, then that field will be of either `Optional` or `OptionalInt`. For 
+example, `setInheritance` in the collections endpoint has an `admin` field that takes an `OptionalInt`: 
+```java
+// Don't want to set the field: 
+client.collections.setInheritance("token", "lpath", 1, OptionalInt.empty());
+
+// Want to set the field to 1:
+client.collections.setInheritance("token", "lpath", 1, OptionalInt.of(1));
+```
+If the operation request includes more than one optional fields, the method will accept a params object that you pass as an 
+instance of a specific class containing those fields. You can set only the desired optional fields as needed. The name 
+of this class follows the pattern `endpointMethodParams`. If you do not need to pass any optional parameters, you can 
+simply pass `null`. For example, the `remove` operation in the collections endpoint has two optional fields: `noTrash` and `recurse`:
+
+
+```java
+// Want to set the noTrash field. Since remove() has more than one optional field, we will need to instantiate a class 
+// for that optional field.
+CollectionsRemoveParams removeParams = new CollectionsRemoveParams();
+removeParams.setNoTrash(1);
+client.collections().remove(token, lpath, removeParams);
+
+// Not setting any optional fields: 
+client.collections().remove(token, lpath, null);
+```
+
+
+
+## Usage Example 
+The following shows a demonstration on how to use the client to create and remove a collection.
+
+```java
+String baseUrl = "http://11.11.111.111:9000/irods-http-api/0.3.0";
+
+// Create an instance of the IrodsHttpClient with the specified base URL.
+IrodsHttpClient client = new IrodsHttpClient(baseUrl);
+
+// Declare a variable to hold the response from the HTTP request. The Response object holds the HTTP status code
+// and the body of the response. Can be retrieved by using .getHttpStatusCode() and .getBody(), respectively.
+Response res;
+
+// Authenticate a rodsadmin user with the username `rods` and password `rodspass`. An authentication request
+// has a response body containing the bearer token that can be used to execute operations as the authenticated user.
+res = client.authenticate("rods", "rodspass");
+String token = res.getBody(); // Set a token variable to the token returned from the request.
+
+// We will create a collection. We don't want to use the createIntermediates field, so we pass it an empty OptionalInt
```

Comment:
Long lines in a code block will cause GitHub to add a horizontal scrollbar. Break the long lines across multiple lines to avoid the scrollbar. It just makes reading the usage example more convenient.

You can verify the presence of a horizontal scrollbar by simply viewing the branch or commit in your fork on GitHub.

Reply:
Ok, sounds good.